dts/bindings/qspi/nxp,s32-qspi-sfp-frad.yaml

```diff
@@ -0,0 +1,56 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP S32 Quad Serial Peripheral Interface (QSPI) Secure Flash Protection SFP FRAD.
+
+  The SFP FRAD performs second-level checks on input flash write and erase transactions,
+  based on the address range of each transaction.
+
+compatible: "nxp,s32-qspi-sfp-frad"
+
+include: base.yaml
+
+properties:
+  start-address:
+    type: int
+    required: true
+    description: |
+      Specifies the specific flash memory region start address.
+
+  end-address:
+    type: int
+    required: true
+    description: |
+      Specifies the specific flash memory region end address.
```

Comment:
There is a problem prevent me, the QSPI node has multiple child nodes with different compatible bindings (device flash nodes). In some cases, they all use reg but it is possible dupliacte, which triggers a duplicate reg warning. To avoid this, I decided to use the reg property only for the device flash node.
Do you have any suggestions on how to handle this?@nordicjm

Reply:
I don't know what you mean? You can have a reg inside of a reg, that's how flash partitions work

Reply:
use-case:
```
&qspi0 {
	sfp_frad_0: sfp_frad@0 {
		compatible = "nxp,s32-qspi-sfp-frad";
		reg = <0x0 DT_SIZE_M(512)>;
	};

	s26hs512t: s26hs512t@0 {
		compatible = "nxp,s32-qspi-hyperflash";
		reg = <0>;
		status = "okay";

		partitions {
			compatible = "fixed-partitions";
			#address-cells = <1>;
			#size-cells = <1>;

			storage_partition: partition@0 {
				label = "storage";
				reg = <0x0 0x10000>;
			};
		};
	};
```
the log warning:
![image](https://github.com/user-attachments/assets/ecf79768-8a4a-4e25-9eb1-1d5558b76736)

Reply:
I found solution for this, move `sfp_frad_0` into a parent node, look like as:
```
&qspi0 {
	sfp {
		compatible = "secure-flash-protection";
		#address-cells = <1>;
		#size-cells = <1>;

	        frad_0: frad@0 {
		        compatible = "nxp,s32-qspi-sfp-frad";
		        reg = <0x0 DT_SIZE_M(512)>;
	        };
        };

	s26hs512t: s26hs512t@0 {
		compatible = "nxp,s32-qspi-hyperflash";
		reg = <0>;
		status = "okay";

		partitions {
			compatible = "fixed-partitions";
			#address-cells = <1>;
			#size-cells = <1>;

			storage_partition: partition@0 {
				label = "storage";
				reg = <0x0 0x10000>;
			};
		};
	};
```